layers/vk_layer_utils.h

```diff
@@ -178,6 +178,21 @@ static inline int MostSignificantBit(uint32_t mask) {
 #endif
 }
 
+// Iterates over all set bits and calls the callback with a bit mask corresponding to each flag.
+// FlagBits and Flags follow Vulkan naming convensions for flag types.
+// An example of a more efficient implementation: https://lemire.me/blog/2018/02/21/iterating-over-set-bits-quickly/
+template <typename FlagBits, typename Flags>
+void IterateFlags(Flags flags, std::function<bool(FlagBits flag)> callback) {
```

Comment:
```suggestion
template <typename FlagBits, typename Flags, typename Callback>
void IterateFlags(Flags flags, Callback callback) {
```

I think it is safe to let the compiler decide the best type to choose for `callback`?

Reply:
That's great suggestion. I wanted something like this :)